docs/how-to/test-on-juju-openstack-provider.rst

```diff
@@ -0,0 +1,118 @@
+===============================
+Test on Juju OpenStack provider
+===============================
+
+This document explain how to setup the testing environment on top of the Juju OpenStack provider and how to run COU testing.
+
+
+Bootstrap juju controller on openstack cloud
+--------------------------------------------
+
+Follow the step on the `official document <https://juju.is/docs/juju/manage-controllers#heading--bootstrap-a-controller>`_ to bootstrap a Juju Openstack controller
+
+
+Deploy openstack with juju openstack-provider
+---------------------------------------------
+
+Follow the step on `STSStack Bundles <https://github.com/canonical/stsstack-bundles>`_
+to deploy the openstack cloud
+
+
+.. code::
+
+    git clone https://github.com/canonical/stsstack-bundles.git
+    cd openstack
+
+    # Some of the overlays may be missing here, check all the supported overlays with:
+    ./generate-bundle.sh --list-overlays
+
+    # ussuri-focal is the lowest version that COU supports
+    ./generate-bundle.sh --name some-juju-model-name -r ussuri -s focal --ovn --telemetry
+
+
+(Optional) sshuttle as proxy
+-------------------
+
+If your openstack environment is behind the vpn and you have the bastion server, you can use sshuttle:
+
+.. code:: bash
+
+     sshuttle --ssh-cmd "ssh -i /home/myuser/.ssh/mykey" -v -r ubuntu@your-bastion-ip 10.5.0.0/16
+
+
+Prepare local python environment for COU
```

Comment:
Snap should be the way to go

Reply:
I don't want to limit the user, maybe they are turning something. But I can add some notes here.

Reply:
But you are limiting the user without mentioning the snap part... Anyways, non-blocker. Free to resolve it when you have a conclusion

Reply:
Added